### ISSUE TITLE
MISOCP uses `NLconstraint` or `NLobjective`?

### ISSUE THREAD

**Issue by tasseff** [lanl-ansi/GasModels.jl#189]
This is present in the `v0.7-rc` branch. Trying to solve an MISOCP model with Gurobi or CPLEX, e.g.,
```julia
import GasModels
import Gurobi
import JuMP

gurobi = JuMP.optimizer_with_attributes(Gurobi.Optimizer)
GasModels.run_gf("test/data/matgas/gaslib-40.m", GasModels.MISOCPGasModel, gurobi)
```
will return `ERROR: LoadError: The solver does not support nonlinear problems (i.e., NLobjective and NLconstraint).`

**Comment by rb004f** [comment]
@kaarthiksundar we are seeing this when using Gurobi on the v0.7 branch. Error doesn't occur in the master branch.

**Comment by tasseff** [comment]
This seems to be caused by the use of `NLexpression` in solution definitions, e.g., [here](https://github.com/lanl-ansi/GasModels.jl/blob/bf5ed2eabecd1ae827920bcde65422cb2867d34a/src/core/variable.jl#L376-L377).

**Comment by rb004f** [comment]
Fixed in version 0.7
